### PR TITLE
Internal: Nested Infra new partial widget render. [ED-6413]

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: assets/dev/js/editor/elements/views/base.js [hunk 1]
```diff
@@ -577,11 +587,7 @@ BaseElementView = BaseContainer.extend( {
 		}
 	},
 
-	renderOnChange( settings ) {
-		if ( ! this.allowRender ) {
-			return;
-		}
-
+	renderChanges( settings, shouldRenderUI = true ) {
```
Comment (Nevoss) [inline assets/dev/js/editor/elements/views/base.js:590]: who is passing the arg `shouldRenderUI` and why it is required? (Maybe add explanation in comment)

File: assets/dev/js/editor/elements/views/base.js [hunk 2]
```diff
@@ -619,16 +625,150 @@ BaseElementView = BaseContainer.extend( {
...
+			// To support nested data-binding bypass nested data-binding that are not part of the current.
+			if ( $current.closest( '.elementor-element' ).data( 'id' ) === id ) {
+				if ( this.dataset.bindingType ) {
+					self.dataBindings.push( {
+						el: this,
+						dataset: this.dataset,
+					} );
+				}
+			}
+		} );
```
Comment (Nevoss) [inline assets/dev/js/editor/elements/views/base.js:691]:
```suggestion
		$dataBinding.forEach( ($current) => {
			// To support nested data-binding bypass nested data-binding that are not part of the current.
			if ( 
				$current.closest( '.elementor-element' ).data( 'id' ) === id &&
				this.dataset.bindingType
			) {
				this.dataBindings.push( {
					el: this,
					dataset: this.dataset,
				} );
			}
		} );
```
Maybe can be simplified

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/elements/views/base.js (hunk 1 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/views/base.js]: Handled.

File: assets/dev/js/editor/elements/views/base.js (hunk 2 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/views/base.js:691]: Prefer, as it...

File: assets/dev/js/editor/elements/views/base.js [hunk 3]
```diff
@@ -621,14 +627,148 @@ BaseElementView = BaseContainer.extend( {
...
+	 *
+	 * @param {Object} settings
+	 */
+	renderOnChange( settings ) {
+		if ( ! this.allowRender ) {
+			return;
+		}
+
+		// If not a dynamic change, and renderDataBindings() was successful, then it was rendered.
+		if ( ! settings.__dynamic__ && this.renderDataBindings( settings, this.dataBindings ) ) {
```
Comment (matipojo) [inline assets/dev/js/editor/elements/views/base.js:765]: Maybe it's after removing a dynamic tag, so `settings.__dynamic__` is empty, but it needed to re-render

File: assets/dev/js/editor/elements/views/base.js [hunk 4]
```diff
@@ -621,14 +627,148 @@ BaseElementView = BaseContainer.extend( {
...
+				break;
+
+				case 'content': {
+					if ( undefined !== settings.changed[ dataBinding.dataset.bindingSetting ] ) {
+						const change = settings.changed[ dataBinding.dataset.bindingSetting ];
+
+						changed = true;
+						dataBinding.el.innerHTML = change;
+					}
+				}
```
Comment (matipojo) [inline assets/dev/js/editor/elements/views/base.js:740]: Extract to a function

**Review by matipojo — CHANGES_REQUESTED** [review]
Please cover with tests

File: assets/dev/js/editor/elements/views/base.js [hunk 5]
```diff
@@ -616,14 +622,148 @@ BaseElementView = BaseContainer.extend( {
...
+			// To support nested data-binding bypass nested data-binding that are not part of the current.
+			if ( $current.closest( '.elementor-element' ).data( 'id' ) === id ) {
+				if ( this.dataset.bindingType ) {
+					self.dataBindings.push( {
+						el: this,
+						dataset: this.dataset,
+					} );
+				}
+			}
+		} );
```
Comment (matipojo) [inline assets/dev/js/editor/elements/views/base.js:686]:
```suggestion
			$dataBinding.filter( ( current )  => {

			// To support nested data-binding bypass nested data-binding that are not part of the current.
			if ( jQuery( current ).closest( '.elementor-element' ).data( 'id' ) === id ) {
				if ( this.dataset.bindingType ) {
					self.dataBindings.push( {
						el: current,
						dataset: current.dataset,
					} );
				}
			}
		} );
```

**Review by matipojo — COMMENTED** [review]
Please resolve the conflicts

**Pull request merged**